src/python/pants/core/target_types.py

```diff
@@ -324,6 +324,26 @@ async def package_archive_target(field_set: ArchiveFieldSet) -> BuiltPackage:
     return BuiltPackage(archive, (BuiltPackageArtifact(output_filename),))
 
 
+# -----------------------------------------------------------------------------------------------
+# `gen_rule` target
+# -----------------------------------------------------------------------------------------------
+
+
+class GenRuleCommandField(StringField):
+    alias = "command"
+    required = True
+
+
+class GenRuleTarget(Target):
+    alias = "gen_rule"
```

Comment:
Although we should definitely call out the connection to `gen_rule` in the docs, we should probably also choose a more useful name! I've never really know what the `gen` stands for ("generate"? "generic"?).

Reply:
How about `ext_rule`. `ext` being as diffuse as `gen`, could stand for "external", "extended" etc :P 

No, but more seriously, perhaps it could be called simply `make`. As a target for "making" stuff.. not to be confused with the tool named "make", though, so perhaps not that great any way.. oh, dang. Naming stuff is hard.

Reply:
I think I'm still confused what the goal does. You run arbitrary bash commands and the generated files can be dependended upon by other targets? You can also use it to trigger side effects like starting a server?

Reply:
Yeah, as a way to process files midway through a rule graph. Haven't thought of starting servers or such. Could possibly be used for that, but not sure it is good practice (unclear life cycle management).

Reply:
```python
make(
  name="Mangle my files to proper format",
  dependencies=[":some-inputs", ":util-scripts"],
  output_files=["result.out"],
  tools=["awk", "jq", "sed"],
  command="./format-util *.in > result.out"
)
```

Reply:
Or, just call it `bash`, or `shell` even, as I'm leaning towards always executing the `command` using `bash -c <command>`, and if we want to open up for making what shell to use optional, then a more generic target name like `shell` makes sense to me.

```python
shell(
    ...,
    command="echo 'foo bar' > output",
    outputs=["output", "dest/"],
)
```

Reply:
The plan is to rename `shell_library` to `shell_sources` and `shell_source`. So here, we could maybe have `shell_command`?

>  It seems like taking advantage of that symmetry would require that this target type is able to usefully depend on shell_library targets to source / import them...

I think it makes sense that dependencies should be populated in the context, such that you could indeed set your command to do something like `command=build-support/my_script.sh`.

Reply:
I was hesitant about requiring this at first, but thinking about it more, it definitely aligns with best practice: your actual binary definition should be small, because it's much harder to test a binary than it is to test a library.